core/src/impl/Kokkos_Profiling_Interface.hpp

```diff
@@ -100,21 +100,19 @@ inline ExecutionSpaceIdentifier identifier_from_devid(const uint32_t in) {
   // out.type = in >> 24;
   // out.device_id = in >> 17;
   // out.instance_id = ((uint32_t(-1)) << 17 ) & in;
-  return {devicetype_from_uint32t(in >> 24),
-          (~((uint32_t(-1)) << 24)) & (in >> 17),
-          (~((uint32_t(-1)) << 17)) & in};
+  return {devicetype_from_uint32t(in >> 24),     /*First 8 bits*/
+          (~((uint32_t(-1)) << 7)) & (in >> 17), /*Next 7 bits */
+          (~((uint32_t(-1)) << 17)) & in};       /*Last 17 bits*/
 }
 
 template <typename ExecutionSpace>
 struct DeviceTypeTraits;
 
-constexpr const size_t device_type_bits = 8;
-constexpr const size_t instance_bits    = 24;
 template <typename ExecutionSpace>
 constexpr uint32_t device_id_root() {
   constexpr auto device_id =
       static_cast<uint32_t>(DeviceTypeTraits<ExecutionSpace>::id);
-  return (device_id << instance_bits);
+  return (device_id << 17);
```

Comment:
Magic number: `instance_bits` was easier to read.

Reply:
I agree. I think we should reintroduce `device_type_bits` and `instance_bits` and also use them in `identifier_from_devid`.